core/iwasm/common/wasm_native.c

```diff
@@ -570,6 +570,12 @@ wasm_native_init()
     if (!wasm_native_register_natives("wasi_nn", native_symbols,
                                       n_native_symbols))
         goto fail;
+#if WASM_ENABLE_WASI_EPHEMERAL_NN != 0
+    n_native_symbols = get_wasi_ephemeral_nn_export_apis(&native_symbols);
+    if (!wasm_native_register_natives("wasi_ephemeral_nn", native_symbols,
+                                      n_native_symbols))
+        goto fail;
+#endif /* WASM_ENABLE_WASI_EPHEMERAL_NN != 0 */
```

Comment:
I strongly believe that we shouldn't replicate the code but instead define here `wasi_nn` or `wasi_ephemeral_nn` as part of the configuration.

Reply:
Just FYI, in [wasi-nn](https://github.com/bytecodealliance/wasi-nn), both [rust api](https://github.com/bytecodealliance/wasi-nn/blob/9d825ab99921f92dfdee719e2b6d6aba366501dd/rust/src/generated.rs#L304) and assembly script api are using "wasi_ephemeral_nn" as its one and only import name.

Reply:
Before deprecating `wasi_nn` I'd give the option to select one or the other.

Reply:
> Just FYI, in [wasi-nn](https://github.com/bytecodealliance/wasi-nn), both [rust api](https://github.com/bytecodealliance/wasi-nn/blob/9d825ab99921f92dfdee719e2b6d6aba366501dd/rust/src/generated.rs#L304) and assembly script api are using "wasi_ephemeral_nn" as its one and only import name.

Do you mean `wasi_nn` and `wasi_ephemeral_nn` only need to be registered one and only one?

Reply:
> I strongly believe that we shouldn't replicate the code but instead define here `wasi_nn` or `wasi_ephemeral_nn` as part of the configuration.

It seems there are slight difference in this file between original `wasi_nn_xx()` and newly added `wasi_ephemeral_nn_xx()`. I am wondering can merge both?

Reply:
I'd consider on supporting only 1 at a time and update the implementation of `wasi_nn_xx` and `wasi_ephemeral_nn`, if there are any.